CMakeLists.txt

```diff
@@ -27,6 +27,51 @@ if(NOT CMAKE_BUILD_TYPE)
   set(CMAKE_BUILD_TYPE "Debug")
 endif()
 
+get_filename_component(CMAKE_AR_NAME "${CMAKE_AR}" NAME)
+function(bpftime_add_static_lib_component_command target)
+  if(APPLE)
+    if(CMAKE_AR_NAME STREQUAL "ar")
+      list(APPEND CMDS
+        COMMAND ${CMAKE_COMMAND} -E make_directory objs/${target}
+        COMMAND bash ${CMAKE_SOURCE_DIR}/cmake/extract_and_rename.sh $<TARGET_FILE:${target}> objs/${target}
+      )
+      set(BPFTIME_STATIC_LIB_AR_CMDS ${BPFTIME_STATIC_LIB_AR_CMDS} ${CMDS} PARENT_SCOPE)
+    elseif(CMAKE_AR_NAME STREQUAL "libtool")
```

Comment:
Are we able to pack static archive using libtool?

Reply:
no, shall I remove it? does any system use libtool for static archives? I tried looking for same but only found 'ar' being used. what do you suggest?

Reply:
> no, shall I remove it? does any system use libtool for static archives? I tried looking for same but only found 'ar' being used. what do you suggest?

Remove it if we don't support libtool, and should report an error if `ar` is not found

Reply:
Done.